### DISCUSSION TITLE
Add async variant of `InstallationProxyService`

### DISCUSSION THREAD

**Discussion by alex-opeykin** [doronz88/pymobiledevice3#1193] [Ideas]
It would be great to add an asynchronous implementation for the install, uninstall application methods in the InstallationProxyService.
Unfortunately, I haven't found it. But I see that there are already async methods in `ServiceConnection` that will help to do this.
I could try to add such a thing myself, but I'm afraid my solution won't be very clean)

While searching for a solution, I realized that when we poll connected devices in async_get_tunneld_devices, we call await rsd.connect(). And in `RemoteServiceDiscoveryService`, `async def connect` uses synchronous` start_lockdown_service`, which does not call `service.aio_start()`. Can we safely replace synchronous `start_lockdown_service` in connect method with `aio_start_lockdown_service?`

**Comment by doronz88** [comment]
In general, the move to asyncio is much more preferred. However, the code duplication this implementation creates can lead to unmaintable code. We plan to refactor the whole project to asyncio one day - which we'll be a major version bump.

So, either wait for this to happen which won't be any time soon, or create a PR that refactors the whole lockdown services implementation to be pure asyncio.
